csrc/scheduler/reduction.cpp

```diff
@@ -556,38 +605,33 @@ std::shared_ptr<ReductionParams> outerReductionHeuristic(
 
   // If there's available parallelism, divide it between threads, blocks, and
   // vectorization
-  // Threads are currently at a warp (16 or 32)
+  // Threads are currently at min_warp_size (8)
   // Blocks are currently at a quarter wave
   // Unroll is currently at 1
   while (
       // and there's parallelism left
       available_parallelism() > 1 &&
       (
           //  There's a place to put it in the block
-          target_threads_in_block <
-              ceilDiv(device_max_threads_per_multiprocessor, (int64_t)4)
+          target_threads_in_block * 2 <= max_target_threads_in_block
```

Comment:
since `target` starts from `8(new) or 16 (old)`, it will reach `512`, there is no actual change using `if(target < max)` or `if(target * 2 <= max)` when `max = 512`

Reply:
Sounds fair. Thanks for walking me through this. :bow: